packages/go_router/lib/src/information_provider.dart

```diff
@@ -149,8 +142,8 @@ class GoRouteInformationProvider extends RouteInformationProvider
 
   void _setValue(String location, Object state) {
     final bool shouldNotify =
-        _value.location != location || _value.state != state;
-    _value = RouteInformation(location: location, state: state);
+        _value.uri.path != location || _value.state != state;
```

Comment:
same here this should also account for query parameter and hashfragment

Reply:
Same as above ⬆️, believe these have been fixed but let me know if you think otherwise.